### ISSUE TITLE
To discuss; Rule 'AvoidDuplication'

### ISSUE THREAD

**Issue by JohnCornegge** [Omnext/omnext.github.io#11]
The rule 'AvoidDuplication' generates a lot of failed items on exclusive splits where there a checks if a previously retrieved object is not empty.
In a microflow, these checks often occur more than once. Currently, the rule does not seem to check which object is being tested for 'empty'. Therefore, the first check would be valid and any other check will fail.

Suggested behavior; If the object that is being checked for empty is different from a previous check, it should not generate a failed item.

**Comment by pimvdnoll** [comment]
I agree on this one. We do a lot tests on $var != empty in microflows. As long $var is not checked multiple times there is no duplication.
